webapp/src/components/BidAssetPage/BidAssetPage.container.js

```diff
@@ -0,0 +1,100 @@
+import { withRouter } from 'react-router'
+import { connect } from 'react-redux'
+import { push } from 'react-router-redux'
+
+import { locations } from 'locations'
+import { ASSET_TYPES } from 'shared/asset'
+import { buildCoordinate } from 'shared/coordinates'
+import { bidRequest } from 'modules/bid/actions'
+import { openModal } from 'modules/ui/actions'
+import {
+  getMatchParams,
+  getMatchParamsCoordinates
+} from 'modules/location/selectors'
+import { getWallet, isConnected, isConnecting } from 'modules/wallet/selectors'
+import { isLoading, getAuthorizations } from 'modules/authorization/selectors'
+import { isBidIdle, getBidByAssetIdFactory } from 'modules/bid/selectors'
+import BidAssetPage from './BidAssetPage'
+
+const mapState = (state, ownProps) => {
+  let assetId
+  switch (ownProps.assetType) {
+    case ASSET_TYPES.parcel: {
+      const { x, y } = getMatchParamsCoordinates(ownProps)
+      assetId = buildCoordinate(x, y)
+      break
+    }
+    case ASSET_TYPES.estate: {
+      const { id } = getMatchParams(ownProps)
+      assetId = id
+      break
+    }
+    default:
+      break
+  }
+
+  const wallet = getWallet(state)
+
+  let authorization
+
+  if (wallet) {
+    authorization = getAuthorizations(state)
+  }
+
+  const getBidByAssetId = getBidByAssetIdFactory(ownProps.assetType, assetId)
+
+  return {
+    wallet,
+    authorization,
+    id: assetId,
+    bid: getBidByAssetId(state),
+    assetType: ownProps.assetType,
+    isTxIdle: isBidIdle(state),
+    isConnected: isConnected(state),
+    isLoading: isConnecting(state) || isLoading(state)
+  }
+}
+
+const mapDispatch = (dispatch, ownProps) => {
+  let onCancel
+  let onBid
```

Comment:
**IF** you want to abstract onBid, you could do:

```js
  const getOnBid = assetType => bid =>
    dispatch(
      openModal('FatfingerModal', {
        assetType,
        priceToConfirm: bid.price,
        onSubmit: () => dispatch(bidRequest({ ...bid, asset_type: assetType }))
      })
    )
```

`onBid = getOnBid(ASSET_TYPES.parcel)`

only if you feel it'll help readability